File: DemoMMOUnity/Lua/Game/UI/CommonUI/LoadingRotate.lua
--- 控制 网络 发送时 的 警示动画
--04

LoadingRotate = class("LoadingRotate")

function LoadingRotate:OnInit(view)
	self.gameObject = view.gameObject
	self.loadingDot = view.loadingDot
end

function LoadingRotate:OnShow()
	self.gameObject:SetActive(true)
	self.loadingDot.enabled = true
end
function LoadingRotate:OnClose()
	self.loadingDot.enabled = false
	self.gameObject:SetActive(false)
end

File: UnityMMOClient/Assets/Game/ZJYFrameWork/Script/Common/LoadingRotate.cs
using System;
using UnityEngine;

namespace ZJYFrameWork.UISerializable.Common
{
    public class LoadingRotate : MonoBehaviour
    {
        public Animation loadingDot;

        public void OnShow()
        {
            gameObject.SetActive(true);
            loadingDot.enabled = true;
        }

        public void OnClose()
        {
            gameObject.SetActive(false);
            loadingDot.enabled = false;
        }

        private void OnDisable()
        {
            loadingDot.enabled = false;
        }
    }
}